app/scripts/controllers/import.js

```diff
@@ -0,0 +1,366 @@
+'use strict';
+
+angular
+  .module('wcagReporter')
+  .controller('ImportCtrl', function (
+    fileReader,
+    $scope,
+    $rootScope,
+    evalContextV3,
+    evalModel,
+    types
+  ) {
+    var JSONLD = window.jsonld;
+    var FEEDBACK = {
+      ERROR: {
+        type: 'danger',
+        message: 'Import error'
+      },
+      PENDING: {
+        type: 'info',
+        message: 'Import pending...'
+      },
+      SUCCESS: {
+        type: 'success',
+        message: 'Import successfull'
+      }
+    };
+
+    $scope.assertionImport = [];
+
+    $scope.allowedMime = [
+      'application/json',
+      'application/ld+json'
+    ].join(',');
+
+    $scope.feedback = false;
+    $scope.importFile = undefined;
+    $scope.importConfirmed = undefined;
+
+    /**
+     * Assertions that get imported need to be validated against
+     * 1. test: should be directly known / related to WCAG
+     * 2. subject should be related to one of the samples
+     * 3. result: being an earl:TestResult
+     * 4. assertedBy: Nice to know who / what made this assertion
+     * @param  {earl:Assertion} assertion [description]
+     * @return {boolean}           validity
+     */
+    function isValidAssertion (assertion) {
+      function hasRequiredKeys (_assertion) {
+        var assertionKeys = Object.keys(_assertion);
+        var requiredKeys = [
+          'test',
+          'subject',
+          'result',
+          'assertedBy'
+        ];
+
+        var missingKeys = [];
+        var key;
+
+        for (key in requiredKeys) {
+          if (assertionKeys.indexOf(requiredKeys[key]) === -1) {
+            missingKeys.push(requiredKeys[key]);
+          }
+        }
+
+        if (missingKeys.length > 0) {
+          return false;
+        }
+
+        return true;
+      }
+
+      function isSampleRelated (subject) {
+        var sampleUrls = evalModel.sampleModel.getPages()
+          .map(function getUrls (page) {
+            var pageUrl;
+
+            if (page.source !== undefined) {
+              try {
+                pageUrl = new URL(page.source);
+              } catch (e) {
+                console.error(e);
+
+                return page.source;
+              }
+
+              return pageUrl.href;
+            }
+          });
+
+        var subjectUrl = '';
+
+        if (typeof subject === 'string') {
+          try {
+            subjectUrl = new URL(subject).href;
+          } catch (e) {
+            console.error('Expected valid url in import assertion subject.');
+
+            return false;
+          }
+        }
+
+        if (
+          typeof subject === 'object' &&
```

Comment:
- [ ] Create an isObjectLiteral helper (Most likely this check is used at other places)